### PR TITLE
Datastore

### PR DESCRIPTION
**Pull request by NoyFlat**
Add the servlet that stores the markers in datastore and  fetches the markers from datastore

### CODE REVIEW

File: capstone-project/src/main/java/com/google/sps/data/Marker.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            this.reporter = reporter;
+            return this;
+        }
+
+        public Builder setDescription(String description){
+            this.description = description;
+            return this;
+        }
+
+        public Marker build(){
```
Comment (NaamaHolzer) [inline capstone-project/src/main/java/com/google/sps/data/Marker.java:56]: I think this constructor should be private since we have the builder.

**Review by NaamaHolzer — APPROVED** [review]

File: capstone-project/src/main/java/com/google/sps/servlets/MarkerServlet.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            String reporter = (String) entity.getProperty("reporter");
+            String description = (String) entity.getProperty("description");
+
+            Marker marker = new Marker.Builder()
+                                .setLat(lat)
+                                .setLng(lng)
+                                .setAnimal(animal)
+                                .setReporter(reporter)
+                                .setDescription(description)
+                                .build();
```
Comment (orarch) [inline capstone-project/src/main/java/com/google/sps/servlets/MarkerServlet.java:89]: Consider having a fromEntity method in Marker class.

**Review by NaamaHolzer — APPROVED** [review]

File: capstone-project/src/main/java/com/google/sps/data/Marker.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.Entity;
+
+/** Represents an animal report marker on the map. */
+public class Marker {
```
Comment (orarch) [inline capstone-project/src/main/java/com/google/sps/data/Marker.java:20]: Extra read: 

https://github.com/objectify/objectify

You can try to apply that in a future PR if it doesn't turn up too complicated

Comment (NoyFlat) [reply capstone-project/src/main/java/com/google/sps/data/Marker.java:20]: Thanks! I'm reading about it and we will see if it's useful for us.

**Review by orarch — APPROVED** [review]

**Pull request merged**